### PR TITLE
feat: kvlightbox component was rewritten using composition api and a …

### PR DESCRIPTION
**Pull request by roger-in-kiva**
kvlightbox component was rewritten using composition api and a basic testing file was added

### CODE REVIEW

File: @kiva/kv-components/vue/KvLightbox.vue [hunk 1]
```diff
@@ -200,84 +207,97 @@ export default {
...
+	],
+	setup(props, { emit }) {
+		const {
+			visible,
+			variant,
+			preventClose,
+		} = toRefs(props);
+
+		const kvLightboxBody = ref(null);
+		const controlsRef = ref(null);
```
Comment (emuvente) [inline @kiva/kv-components/vue/KvLightbox.vue:221]: I think these refs will need to be bound in the template. For instance, `ref="kvLightbox"` would change to `:ref="kvLightbox"`

**Review by emuvente — COMMENTED** [review]
This test is failing in vue 3 right now:
```
KvLightbox › has no automated accessibility violations

    TypeError: Cannot read properties of undefined (reading '_c')

       7 | describe('KvLightbox', () => {
       8 | 	it('has no automated accessibility violations', async () => {
    >  9 | 		const { container } = render(KvLightbox,
         | 		                      ^
      10 | 			{
      11 | 				props: { title: 'Lightbox Title' },
      12 | 			});

      at Proxy.render (../node_modules/vue-focus-lock/dist/index.js:921:92)
      at renderComponentRoot (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:465:44)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4274:57)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:164:29)
      at setupRenderEffect (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4400:9)
      at mountComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4183:9)
      at processComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4141:17)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3744:21)
      at mountChildren (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3929:13)
      at mountElement (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3839:17)
      at processElement (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3821:13)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3741:21)
      at mountChildren (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3929:13)
      at mountElement (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3839:17)
      at processElement (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3821:13)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3741:21)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4281:21)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:164:29)
      at setupRenderEffect (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4400:9)
      at mountComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4183:9)
      at processComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4141:17)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3744:21)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4281:21)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:164:29)
      at setupRenderEffect (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4400:9)
      at mountComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4183:9)
      at processComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4141:17)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3744:21)
      at ReactiveEffect.componentUpdateFn [as fn] (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4281:21)
      at ReactiveEffect.run (node_modules/@vue/reactivity/dist/reactivity.cjs.js:164:29)
      at setupRenderEffect (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4400:9)
      at mountComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4183:9)
      at processComponent (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4141:17)
      at patch (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3744:21)
      at render (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:4885:13)
      at mount (node_modules/@vue/runtime-core/dist/runtime-core.cjs.js:3167:25)
      at Object.app.mount (node_modules/@vue/runtime-dom/dist/runtime-dom.cjs.js:1530:23)
      at mount (node_modules/@vue/test-utils/dist/vue-test-utils.cjs.js:8002:18)
      at render (node_modules/@testing-library/vue/dist/render.js:43:38)
      at Object.<anonymous> (local/tests/unit/specs/components/KvLightbox.spec.js:9:25)
      at TestScheduler.scheduleTests (../node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (../node_modules/@jest/core/build/runJest.js:404:19)
      at _run10000 (../node_modules/@jest/core/build/cli/index.js:320:7)
      at runCLI (../node_modules/@jest/core/build/cli/index.js:173:3)
```
You can reproduce this locally by running `npm test --workspace test-vue3 KvLightbox`.

I think to fix this we need to install version 2 of vue-focus-lock, which has updates to support vue 3.

File: @kiva/kv-components/vue/KvLightbox.vue (hunk 1 above)
Comment (roger-in-kiva) [reply @kiva/kv-components/vue/KvLightbox.vue:221]: I though the same, but, 
https://v3.vuejs.org/guide/composition-api-template-refs.html

Your kvbutton has the same.

Comment (emuvente) [reply @kiva/kv-components/vue/KvLightbox.vue:221]: oh, right, good point!

**Review by emuvente — APPROVED** [review]
👍 looks great!

**Pull request merged**